vkcs/compute/resource_instance.go

```diff
@@ -788,7 +789,7 @@ func resourceComputeInstanceUpdate(ctx context.Context, d *schema.ResourceData,
 		if strings.ToLower(powerStateNew) == "active" {
 			if strings.ToLower(powerStateOld) == "shelved" || strings.ToLower(powerStateOld) == "shelved_offloaded" {
 				unshelveOpt := &shelveunshelve.UnshelveOpts{
-					AvailabilityZone: d.Get("availability_zone").(string),
+					AvailabilityZone: d.GetRawConfig().GetAttr("availability_zone").AsString(),
```

Comment:
If you put the mentioned logic with suppressing `requires new` from diff created by `az` changes, then it should be possible to retrieve the value from configuration as usual. This attribute, I think, becomes empty due to your manipulation of the planned value.

Reply:
done